### PR TITLE
fix(plot): disable singledispatchmethod in plot_facade

### PR DESCRIPTION
**Pull request by atsushi421**
## Description
This PR disables `singledispatchmethod` library in plot_facade.py.

## Related links
- Discussion: https://tier4.atlassian.net/browse/RT2-713?focusedCommentId=122819
- To reproduce: 
  - [ipynb, html](https://drive.google.com/drive/folders/1rqbZhhfK1Q_kv7D5AXBza_3D_84lZQkT?usp=share_link)
  - [trace data](https://drive.google.com/file/d/1xy1L4c-eGi9RSHog1qNgovTcwxoM74Z9/view?usp=share_link)
  - [architecture file](https://drive.google.com/file/d/1W9PDyj046xzapMA-p3o6HWtUhMA8JhIG/view?usp=share_link)

## Notes for reviewers
There is no degradation using CARET_report.
image

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] (Optional) The PR has been properly tested with CARET_report verification.
- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

**Review by rokamu623 — APPROVED** [review]
LGTM!

**Comment by atsushi421** [comment]
@takeshi-iwanari 
Could you please review this PR?

File: src/caret_analyze/plot/plot_facade.py [hunk 1]
```diff
@@ -29,40 +27,42 @@
...
+    Parse target argument.
 
-    @singledispatchmethod
-    def create_period_timeseries_plot(
-        target_objects: Collection[TimeSeriesTypes]
-    ) -> PlotBase:
-        """
-        Get period timeseries plot instance.
+    To address both cases where the target argument is passed in collection type
+    and unpacked, this function converts them to the same list format.
```
Comment (nabetetsu) [inline src/caret_analyze/plot/plot_facade.py:42]: Strictly speacking , "collection type or unpacked" ?

File: src/caret_analyze/plot/plot_facade.py [hunk 2]
```diff
@@ -113,34 +125,20 @@ def create_frequency_timeseries_plot(
...
         """
         Get latency timeseries plot instance.
 
         Parameters
         ----------
-        target_object : TimeSeriesTypes
-            TimeSeriesPlotTypes = Union[
-                CallbackBase, Communication, Union[Publisher, Subscription]
-            ]
+        target_object : Collection[CallbackBase | Communication]
```
Comment (nabetetsu) [inline src/caret_analyze/plot/plot_facade.py:141]: target_objects ?

**Review by nabetetsu — COMMENTED** [review]
@atsushi421 I think the code fix is perfect.
It is not directly related to the modification, but there are some points of concern in the comment, could you please check it out just in case?

---
The function def create_response_time_stacked_bar_plot() is missing docstring in public method.

---
167:         path : Collection[Path]
                -> paths

---
184:        response time calculation method, by default best-to-worst.
                -> Response time calculation method, best-to-worst by default.

---
203:       Start time of cropping range, by default 0.
              -> Start time of the cropping range, 0 by default.

---
205:        End point of cropping range, by default 0.
               -> End point of cropping range, 0 by default.

---
236:        Granularity of chain with two value; [raw/node], by default None.
               -> Granularity of the chain with two values; [raw/node], None by default.

---
239:         by default False.
                -> False by default.

---
241:         Start time of cropping range, by default 0.
                -> Start time of cropping range, 0 by default.

---
243:         End point of cropping range, by default 0
                -> End point of cropping range, 0 by default.

**Comment by atsushi421** [comment]
@nabetetsu 
Thank you for your careful review. I have made the corrections, could you please check it again?

**Comment by nabetetsu** [comment]
LGTM :+1:

**Review by nabetetsu — APPROVED** [review]
LGTM :+1:

**Pull request merged**